Signal/src/ViewControllers/ConversationView/MessagesViewController.m

```diff
@@ -1886,8 +1853,8 @@ - (TSOutgoingMessage *)nextOutgoingMessage:(NSIndexPath *)indexPath
 {
     NSInteger rowCount = [self.collectionView numberOfItemsInSection:indexPath.section];
     for (NSInteger row = indexPath.row + 1; row < rowCount; row++) {
-        id<OWSMessageData> nextMessage =
-            [self messageAtIndexPath:[NSIndexPath indexPathForRow:row inSection:indexPath.section]];
+        NSIndexPath *nextIndexPath = [NSIndexPath indexPathForRow:row inSection:indexPath.section];
+        TSInteraction *nextMessage = [self interactionAtIndexPath:nextIndexPath];
```

Comment:
This fixes an unrelated and embarrassing bug - we need to check the type of the _message (interaction)_, not the _message adapter_.